src/rascal/structure_managers/structure_manager.hh

```diff
@@ -1281,31 +1316,32 @@ namespace rascal {
 
     //! returns the counters - which is the position in a list at each
     //! Order.
-    std::array<size_t, Order> get_counters() {
-      std::array<size_t, Order> counters;
-      counters[Order - 1] = this->index;
-      if (Order == 1) {
-        return counters;
-      } else {
-        auto parental_counters = this->container.get_counters();
-        for (size_t i{0}; i < Order - 1; i++) {
-          counters[i] = parental_counters[i];
-        }
-        return counters;
-      }
+    template <bool T = IsOrderOne, std::enable_if_t<T, int> = 0>
+    std::array<size_t, 1> get_counters() {
+      std::array<size_t, 1> counters{this->index};
+      return counters;
     }
-    std::array<size_t, Order> get_offsets() {
-      std::array<size_t, Order> offsets;
-      offsets[Order - 1] = this->offset;
-      if (Order == 1) {
-        return offsets;
-      } else {
-        auto parental_offsets = this->container.get_offsets();
-        for (size_t i{0}; i < Order - 1; i++) {
-          offsets[i] = parental_offsets[i];
-        }
-        return offsets;
-      }
+
+    template <bool T = IsOrderOne, std::enable_if_t<not(T), int> = 0>
+    std::array<size_t, 2> get_counters() {
+      auto && parental_counters = this->container.get_counters();
+      std::array<size_t, 2> counters{parental_counters[0], this->index};
+      return counters;
+    }
+
+    //! get offsets at which the current clusters at several orders should
+    //! start
+    template <bool T = IsOrderOne, std::enable_if_t<T, int> = 0>
```

Comment:
Please don't use `T` for a boolean value, since it usually refers to a type. Do you need the possibility to call this function with the first parameter being different from `IsOrderOne`? If not, `template <std::enable_if_t<IsOrderOne, int> = 0>` should be enough, else you can use `bool is_order_one = IsOrderOne` or something similar.

Reply:
There are a lot of them, so I am not going to mark every single one

Reply:
template <std::enable_if_t<IsOrderOne, int> = 0> does not lead to SFINAE. your suggestion of bool is_order_one = IsOrderOne is really too lengthy for no reason in my opinion. I like my template lines to fit on one line.
T is often used to refer to types but I think that having bool T is quite clear on the fact that this is a bool isn't it ? Would bool val be good for you ? 
Keep in mind that this is not a variable to be used anywhere else than in the template section.

Reply:
Sorry I missed this one too. How about `bool condition = XXX` or even `bool cond = XXX`?